src/impl_/pyclass/lazy_type_object.rs

```diff
@@ -46,15 +46,30 @@ impl<T> LazyTypeObject<T> {
 
 impl<T: PyClass> LazyTypeObject<T> {
     /// Gets the type object contained by this `LazyTypeObject`, initializing it if needed.
+    #[cfg_attr(
+        not(feature = "gil-refs"),
+        deprecated(
+            since = "0.21.0",
+            note = "`LazyTypeObject::get_or_init` will be replaced by `LazyTypeObject::get_or_init_bound` in a future PyO3 version"
+        )
+    )]
     pub fn get_or_init<'py>(&'py self, py: Python<'py>) -> &'py PyType {
+        self.get_or_init_bound(py).as_gil_ref()
+    }
+
+    /// Gets the type object contained by this `LazyTypeObject`, initializing it if needed.
+    pub fn get_or_init_bound<'py>(&'py self, py: Python<'py>) -> &Bound<'py, PyType> {
         self.get_or_try_init(py).unwrap_or_else(|err| {
             err.print(py);
             panic!("failed to create type object for {}", T::NAME)
         })
     }
 
     /// Fallible version of the above.
-    pub(crate) fn get_or_try_init<'py>(&'py self, py: Python<'py>) -> PyResult<&'py PyType> {
+    pub(crate) fn get_or_try_init<'py>(
+        &'py self,
```

Comment:
I believe the restriction on `self` is no longer necessary; that was a constraint forced by the way the GIL Ref API mixed two lifetimes into one.

```suggestion
        &self,
```

Some of the rest of these functions can probably be changed in the same way.